lua/tardis/sh_screens.lua

```diff
@@ -146,6 +146,13 @@ end
 
 function TARDIS:RemoveHUDScreen()
 	if IsValid(self.screenpopframe) then
+		local curscreen = self.screenpop.curscreen
+		if IsValid(curscreen) and curscreen.OnCloseScreen then
+			local result = curscreen:OnCloseScreen()
+			if not result then
```

Comment:
change this so that only explicitly returning false cancels the screen removal, then you don't need to return true in the OnCloseScreen thing in the destinations screen, just return nil, but otherwise good stuff this is how i'd have done it